### PR TITLE
feat: Add handoff to user tool

### PR DESCRIPTION
**Pull request by mkmeral**
## Description

This PR introduces a new handoff_to_user tool that enables agents to hand off control to users for human intervention in automated workflows. The tool provides two distinct modes of operation:

1. Interactive Mode (breakout_of_loop=False): Pauses execution to collect user input and continues with the response
2. Complete Handoff Mode (breakout_of_loop=True): Stops the event loop entirely and transfers control to the user

The tool follows the same architectural patterns as the existing stop tool, using ToolUse and ToolResult types with direct request_state access for event loop control. It provides a professional user interface with clear visual indicators and comprehensive error handling including graceful recovery from KeyboardInterrupt and other exceptions.

## Related Issues
[No specific issues referenced - this is a new feature implementation]

## Documentation PR
Will do in a separate CR 

## Type of Change
• [ ] Bug fix
• [x] New Tool
• [ ] Breaking change
• [ ] Other (please describe):

## Testing

The implementation has been thoroughly tested with comprehensive test coverage:

### Test Coverage (9 test cases):
• Direct function calls with ToolUse objects (matching stop tool pattern)
• Event loop control verification (stop_event_loop flag setting)
• User input collection and response handling
• Error handling (KeyboardInterrupt, general exceptions)
• Default parameter behavior and edge cases
• Agent interface integration testing
• Tool specification structure validation

### Code Quality Checks:
• hatch fmt --linter ✅
• hatch fmt --formatter ✅ 
• hatch test --all ✅ (only a2a test fails, not related to changes)

### Additional Testing:
```
>>> from strands_tools import handoff_to_user
>>> agent = Agent(tools=[handoff_to_user, calculator])
>>> agent("Ask user to pick a number and then get the square root of it")
I'd be happy to help you with this task. I'll need to ask the user to pick a number, and then I'll calculate its square root using the calculator tool.
Tool #1: handoff_to_user

============================================================
🤝 AGENT REQUESTING USER HANDOFF
============================================================

Please pick a number that you'd like me to find the square root of.

⏳ Waiting for your response...
----------------------------------------
Your response: 29931
============================================================

Thank you for choosing the number 29931. I'll calculate its square root now.
Tool #2: calculator
The square root of 29931 is 173.0057802498.AgentResult(stop_reason='end_turn', message={'role': 'assistant', 'content': [{'text': 'The square root of 29931 is 173.0057802498.'}]}, metrics=EventLoopMetrics(cycle_count=3, tool_metrics={'handoff_to_user': ToolMetrics(tool={'toolUseId': 'tooluse_cGPUkvDUTh2-8sNwlRVPjw', 'name': 'handoff_to_user', 'input': {'message': "Please pick a number that you'd like me to find the square root of.", 'breakout_of_loop': False}}, call_count=1, success_count=1, error_count=0, total_time=6.689070224761963), 'calculator': ToolMetrics(tool={'toolUseId': 'tooluse_onFtuqpHSWWFFcwmNgCrCA', 'name': 'calculator', 'input': {'expression': 'sqrt(29931)', 'mode': 'evaluate'}}, call_count=1, success_count=1, error_count=0, total_time=0.10299992561340332)}, cycle_durations=[1.529723882675171], traces=[<strands.telemetry.metrics.Trace object at 0x10c5c0200>, <strands.telemetry.metrics.Trace object at 0x10c5c03b0>, <strands.telemetry.metrics.Trace object at 0x10c5c02c0>], accumulated_usage={'inputTokens': 6645, 'outputTokens': 241, 'totalTokens': 6886}, accumulated_metrics={'latencyMs': 6498}), state={})
```

## Checklist
• [x] I have read the CONTRIBUTING document
• [x] I have added tests that prove my fix is effective or my feature works
• [] I have updated the documentation accordingly
• [x] I have added an appropriate example to the documentation to outline the feature
• [x] My changes generate no new warnings
• [x] Any dependent changes have been merged and published

• By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your 
choice.

### CODE REVIEW

File: src/strands_tools/handoff_to_user.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    tool_use_id = tool["toolUseId"]
+    tool_input = tool["input"]
+    request_state = kwargs.get("request_state", {})
+
+    # Extract parameters
+    message = tool_input.get("message", "Agent requesting user handoff")
+    breakout_of_loop = tool_input.get("breakout_of_loop", False)
+
+    # Display handoff notification
+    print("\n" + "=" * 60)
```
Comment (arronbailiss) [inline src/strands_tools/handoff_to_user.py]: use `console.print` instead (https://github.com/search?q=repo%3Astrands-agents%2Ftools%20print&type=code)

File: src/strands_tools/handoff_to_user.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            "status": "success",
+            "content": [{"text": f"Agent handoff completed. Message displayed to user: {message}"}],
+        }
+    else:
+        # Wait for user input and continue
+        print("⏳ Waiting for your response...")
+        print("-" * 40)
+
+        try:
+            user_response = input("Your response: ").strip()
```
Comment (arronbailiss) [inline src/strands_tools/handoff_to_user.py]: use `get_user_input` from https://github.com/strands-agents/tools/blob/3390e567012c171a7ee0ec795bebdeb7c1885053/src/strands_tools/utils/user_input.py#L51

**Review by awsarron — CHANGES_REQUESTED** [review]

**Comment by arronbailiss** [comment]
Related issue: https://github.com/strands-agents/sdk-python/issues/214
